src/main/java/Parser.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * <h1>Parser class</h1>
+ * This class helps to parse the data that users input
+ * and separate the different key points in each line
+ * of input, making it easier for the program to decide
+ * which command action to call.
+ */
+public class Parser {
+    private String line;
+    private String commandType = "";
+    private String taskDetails = "";
+    private String date = "";
+
+    public Parser(String line) {
+        this.line = line;
+    }
+
+    /**
+     * Parses the current line of input and checks which command it falls under.
+     * Helps separate out key details in the input such as the task details and date.
+     * @throws DukeException When user input is wrong or incomplete.
+     */
+    public void parse() throws DukeException {
```

Comment:
Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle. - https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html